src/main/java/pl/sda/refactoring/service/OrderService.java

```diff
@@ -0,0 +1,156 @@
+package pl.sda.refactoring.service;
+
+import lombok.NonNull;
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+import pl.sda.refactoring.entity.Currency;
+import pl.sda.refactoring.entity.Order;
+import pl.sda.refactoring.entity.OrderItem;
+import pl.sda.refactoring.entity.OrderStatus;
+
+import java.math.BigDecimal;
+import java.time.Instant;
+import java.time.LocalDate;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.UUID;
+
+import static java.math.RoundingMode.HALF_UP;
+import static java.util.UUID.randomUUID;
+
+@Service
+@Transactional
+public class OrderService {
+
+    private final DiscountService discountService;
+    private final CurrencyService currencyService;
+    private final CustomerService customerService;
+    private final OrderRepo orderRepo;
+    private final EmailService emailService;
+    private final List<LocalDate> freeDeliveryDays;
+    private final List<String> emailRecipients;
+
+    public OrderService(@NonNull DiscountService discountService,
+                        @NonNull CurrencyService currencyService,
+                        @NonNull CustomerService customerService,
+                        @NonNull OrderRepo orderRepo,
+                        @NonNull EmailService emailService,
+                        @NonNull OrderSettings settings) {
+        this.discountService = discountService;
+        this.currencyService = currencyService;
+        this.customerService = customerService;
+        this.orderRepo = orderRepo;
+        this.emailService = emailService;
+        this.freeDeliveryDays = settings.discountSettings().freeDeliveryDays();
+        this.emailRecipients = settings.emailRecipients();
+    }
+
+    public UUID makeOrder(@NonNull UUID customerId,
+                          @NonNull List<OrderItem> orderItems,
+                          @NonNull Currency currency,
+                          String coupon) {
+        boolean exists = customerService.exists(customerId);
+        if (exists) {
```

Comment:
Consider using validator for customer instead of creating this additional nesting.